src/printf.rs

```diff
@@ -307,6 +310,36 @@ mod test {
         assert_eq!(s, r"+101");
     }
 
+    #[test]
+    fn test_sprintf_array() {
+        let values: Vec<Value> = vec!["hello".into(), "world".into()];
+        let s = sprintf("foo %v", &[Value::Array(values)]);
+        assert!(s.is_ok());
+        let s = s.unwrap();
+        assert_eq!(s, r"foo [hello world]");
+
+        let values: Vec<Value> = vec![42.into(), 100.into()];
+        let s = sprintf("foo %v", &[Value::Array(values)]);
+        assert!(s.is_ok());
+        let s = s.unwrap();
+        assert_eq!(s, r"foo [42 100]");
+    }
+
+    #[test]
+    fn test_sprintf_map() {
+        let mut values: HashMap<String, Value> = HashMap::new();
+        values.insert("hello".into(), "world".into());
+        values.insert("number".into(), 42.into());
+        let s = sprintf("foo %v", &[Value::Map(values)]);
+        assert!(s.is_ok());
+        let s = s.unwrap();
+        // The print order is unpredictable, we can't write
+        // a straight comparison
+        assert!(s.contains("hello:world"));
+        assert!(s.contains("number:42"));
+        assert!(Regex::new(r"^foo \{.*\}$").unwrap().is_match(&s));
```

Comment:
We could check `starts_with("foo {")` and `ends_with("}")` to skip the regex dependency.